src/measure_theory/integral/exp_decay.lean

```diff
@@ -0,0 +1,221 @@
+  (continuous_const.mul continuous_id).exp.integrable_on_Ioc
+
+lemma exp_neg_hasderiv (b : ℝ) (x : ℝ) (h : 0 < b) :
+  has_deriv_at (λ (y : ℝ), -exp (-b * y) / b) (exp (-b * x)) x :=
+begin
+  have : has_deriv_at (λ (x : ℝ), x) 1 x := has_deriv_at_id x,
+  have : has_deriv_at (λ (x : ℝ), b*x) b x := by simpa using has_deriv_at.const_mul b this,
+  have : has_deriv_at (λ (x : ℝ), -b*x) (-b) x := by simpa using has_deriv_at.neg this,
+  have : has_deriv_at (λ (x : ℝ), exp (-b*x)) (exp (-b*x)*(-b)) x := has_deriv_at.exp this,
+  have : has_deriv_at (λ (x : ℝ), -exp (-b*x)) (-(exp (-b*x)*(-b))) x := has_deriv_at.neg this,
+  have := (has_deriv_at.div_const this b),
+  have s : (-(exp (-b * x) * -b) / b) = exp(-b*x),
+  { simp only [neg_mul, mul_neg, neg_neg],
+    rw mul_div_cancel _ (h.ne') },
+  rw s at this, exact this
+end
+
+lemma exp_neg_antideriv {b: ℝ} (h: 0 < b):
+  deriv (λ (x : ℝ), -exp (-b * x) / b) = (λ (x : ℝ), exp (-b * x)) :=
+begin
+  ext1,
+  exact has_deriv_at.deriv ( exp_neg_hasderiv b x h )
+end
+
+/-- Integral of `exp(-b x)` over `(a, X)` is bounded as `X → ∞`. -/
+lemma exp_neg_integral_bound {b : ℝ} (a X : ℝ) (h2 : 0 < b):
+  (∫ x in a .. X, exp (-b * x)) ≤ exp(-b*a)/b :=
+begin
+  rw (integral_deriv_eq_sub' (λ x:ℝ, -exp(-b*x)/b ) ),
+  -- goal 1/4: F(X) - F(a) is bounded
+  { simp only [tsub_le_iff_right],
+    rw (neg_div b (exp (-b * a))),
+    rw (neg_div b (exp (-b * X))),
+    rw add_neg_self,
+    rw [neg_le, neg_zero],
+    apply le_of_lt,
+    apply div_pos,
+    apply exp_pos,
+    exact h2 },
+  -- goal 2/4: the derivative of F is exp(-b x)
+  exact exp_neg_antideriv h2,
+  -- goal 3/4: F is differentiable (why isn't this implicit?)
+  intros x hx,
+  exact has_deriv_at.differentiable_at (exp_neg_hasderiv b x h2 ),
+  -- goal 4/4: exp(-b x) is continuous
+  { apply continuous.continuous_on,
+    continuity }
+end
+
+/-- `exp(-b x)` is integrable on `(a, ∞)`. -/
+lemma exp_neg_integrable_Ioi (a : ℝ) {b : ℝ} (h : 0 < b):
+  integrable_on (λ x : ℝ, exp(-b * x)) (Ioi a) :=
+begin
+  apply (integrable_on_Ioi_of_interval_integral_norm_bounded
+    (exp (-b*a)/b) a (exp_neg_finite_integrable b a) tendsto_id),
+  simp only [eventually_at_top, ge_iff_le],
+  use a,
+  intros b2 hb2,
+  have : (λ x:ℝ, ∥exp(-b * x)∥) = (λ x:ℝ, exp(-b * x)),
+  { ext1,apply norm_of_nonneg,
+    exact (exp_pos (-b*x)).le },
+  rw this,
+  apply exp_neg_integral_bound,
+  exact h
+end
+
+lemma exp_neg_finite_integral_Ioi (a : ℝ) {b : ℝ} (h : 0 < b):
+  has_finite_integral (λ x : ℝ, exp(-b * x))
+  (measure_space.volume.restrict (Ioi a)) :=
+begin
+  apply integrable.has_finite_integral,
+  exact exp_neg_integrable_Ioi a h,
+end
+
+/- Now show any continous function which is O(exp(-b x)) is integrable. -/
+
+lemma globalbound_bigoh_exp (f: ℝ → ℝ) (a : ℝ) (b : ℝ) (h : 0 < b)
+  (h1: continuous_on f (Ici a) )
+  (h2: asymptotics.is_O f (λ x:ℝ, exp(-b * x)) at_top) :
+  ∃ (B : ℝ), ∀ (x : ℝ), (x ∈ Ici a) → |f(x)| ≤ B * exp(-b*x)  :=
```

Comment:
I am not sure you need this lemma: in the proof of the next lemma, you could use `integrable_on_union` and apply directly that a continuous function is integrable on an interval of the form `Icc x y`, without needing to write down an explicit bound.

Reply:
You're right, it's not needed -- good point, thanks!